src/schoology-extractor/schoology_extractor/schoology_extract_facade.py

```diff
@@ -122,9 +121,11 @@ def get_sections(self) -> pd.DataFrame:
         # the paging. As BB said in a pr comment, perhaps it should be handled
         # in a decorator. This works but we should consider refactoring to a
         # cleaner approach.
-        sections = pd.DataFrame(
+        sections = sync.sync_resource(
+            RESOURCE_NAMES.SECTION,
+            self._db_engine,
             self._client.get_section_by_course_ids([c["id"] for c in courses_list])
-        )
+            )

```

Comment:
The implementation of sync_resource for further resources, should be as simple as it is here.